app/src/main/java/uk/gov/android/authentication/AppAuthSession.kt

```diff
@@ -0,0 +1,117 @@
+package uk.gov.android.authentication
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import androidx.core.app.ActivityCompat
+import java.util.UUID
+import net.openid.appauth.AuthorizationException
+import net.openid.appauth.AuthorizationRequest
+import net.openid.appauth.AuthorizationResponse
+import net.openid.appauth.AuthorizationService
+import net.openid.appauth.AuthorizationServiceConfiguration
+
+@Suppress("TooGenericExceptionThrown")
+class AppAuthSession : LoginSession {
+    private var context: Context? = null
+    private lateinit var authService: AuthorizationService
+
+    override fun init(
+        context: Context
+    ): LoginSession {
+        if (this.context == null) {
+            this.context = context
+            authService = AuthorizationService(context)
+        }
+        return this
+    }
```

Comment:
Do we not want to provide the context through the constructor?

Reply:
Yep good idea, something to change in another ticket